retro/reco.py

```diff
@@ -425,6 +507,507 @@ def loglike(cube, ndim=None, nparams=None): # pylint: disable=unused-argument
+            #w = rand.uniform(-0.5, 1.5, n_cart)
+            new_x_cart2 = (1 + w) * S_cart[best_idx] - w * new_x_cart
+
+            # first reflect at best point
+            reflected_new_x_spher = np.zeros(n_spher, dtype=spher_cord)
+            reflect(new_x_spher, S_spher[best_idx], reflected_new_x_spher)
+
+            new_x_spher2 = np.zeros_like(new_x_spher)
+
+            # now do a combination of best and reflected point with weight w
+            for dim in ['x', 'y', 'z']:
+                #w = rand.uniform(-0.5, 1.5, n_spher)
+                w = rand.uniform(0, 1, n_spher)
+                new_x_spher2[dim] = (1 - w) * S_spher[best_idx][dim] + w * reflected_new_x_spher[dim]
+            fill_from_cart(new_x_spher2)
+
+            if use_priors:
+                outside = np.any(new_x_cart2 < 0) or np.any(new_x_cart2 > 1)
+            else:
+                outside = False
+
+            if not outside:
+                new_fx = fun(create_x(new_x_cart2, new_x_spher2))
+
+                if new_fx < fx[worst_idx]:
+                    #print('-> MUT accepted')
+                    # found better point
+                    S_cart[worst_idx] = new_x_cart2
+                    S_spher[worst_idx] = new_x_spher2
+                    fx[worst_idx] = new_fx
+                    mutation_success += 1
+                    continue
+
+            '''
+            # random sampling of new point
+
+            # sample new cartesian coordinates from distribution given the livepoints
+            mean_cart = np.average(S_cart, axis=0)
+            cov_cart = np.cov(S_cart.T)
+            new_x_cart = rand.multivariate_normal(mean_cart, cov_cart, 1)[0]
+            # random new angle
+            new_x_spher['az'] = rand.uniform(0,2*np.pi,n_spher)
+            new_x_spher['zen'] = np.arccos(rand.uniform(-1,1,n_spher))
+            fill_from_spher(new_x_spher)
+
+            new_fx = fun(create_x(new_x_cart, new_x_spher))
+
+            if new_fx < fx[worst_idx]:
+                # found better point
+                S_cart[worst_idx] = new_x_cart
+                S_spher[worst_idx] = new_x_spher
+                fx[worst_idx] = new_fx
+                whateverido += 1
+                continue
+            '''
+            
+            # if we get here no method was successful in replacing worst point -> start over
+            failure += 1
+
+
+        res = OrderedDict()
+        res['method'] = 'CRS2spherical+lm+sampling'
+        res['n_live'] = n_live
+        res['max_iter'] = max_iter
+        res['max_noimprovement'] = max_noimprovement
+        res['fn_std'] = fn_std
+        res['use_priors'] = use_priors
+        res['sobol'] = sobol
+        res['stopping_flag'] = stopping_flag
+        res['num_simplex'] = simplex_success
+        res['num_mutation'] = mutation_success
+        res['num_sampling'] = whateverido
+        res['num_failure'] = failure
+        res['num+tot'] = k
+        return res
```

Comment:
again, assuming your indentation is messed up, otherwise this doesn't make sense to have a return and then do stuff below

Reply:
this is because i currently disabled the emcee sampling, didn't help

Reply:
so you disable the emcee code by putting a return statement before the code? you should either put that code a conditional (`if RUN_EMCEE:` where you define that constant at the top) or eliminate the code altogether; it's very confusing to read the code this way, which is why i thought you had indentation issues.

Reply:
(or you can comment out the code but still put the return after the commented-out code, though if it works I'd prefer the conditional so we can easily turn it on again)